File: FoosballGames.WebApi/Controllers/FoosballGamesController.cs
using FoosballGames.Contracts;
using FoosballGames.Infrastructure.Messaging;
using Microsoft.AspNetCore.Mvc;

namespace FoosballGames.WebApi.Controllers;

[ApiController]
[Route("[controller]")]
public class FoosballGamesController : ControllerBase
{
    private readonly ICommandQueryDispatcher _commandQueryDispatcher;

    public FoosballGamesController(ICommandQueryDispatcher commandQueryDispatcher) => 
        _commandQueryDispatcher = commandQueryDispatcher;

    [HttpGet]
    public Task<FoosballGamesResponse> Get() => 
        _commandQueryDispatcher.SendAsync<GetFoosballGames, FoosballGamesResponse>(new GetFoosballGames());

    [HttpGet("{id:Guid}")]
    public Task<Contracts.FoosballGame> GetById(Guid id) => 
        _commandQueryDispatcher.SendAsync<GetFoosballGame, Contracts.FoosballGame>(new GetFoosballGame(id));

    [HttpPost("create-game")]
    public Task Create([FromBody] CreateFoosballGame cmd) => 
        _commandQueryDispatcher.SendAsync(cmd);

    [HttpPost("add-point")]
    public Task AddPoint([FromBody] AddPointForTeam cmd) => 
        _commandQueryDispatcher.SendAsync(cmd);
}

File: FoosballGames.WebApi/Controllers/FoosballGamesController.fs
namespace FoosballGames.WebApi.Controllers

open FoosballGames.Infrastructure.Messaging
open Microsoft.AspNetCore.Mvc
open FoosballGames.Contracts

[<ApiController>]
[<Route("[controller]")>]
type FoosballGamesController(commandQueryDispatcher: ICommandQueryDispatcher) =
    inherit ControllerBase()

    [<HttpGet>]
    member _.Get() =
        commandQueryDispatcher.SendAsync(GetFoosballGames())
    
    [<HttpGet("{id:Guid}")>]
    member _.GetById(id) =
        commandQueryDispatcher.SendAsync({ GetFoosballGame.Id = id })

    [<HttpPost("create-game")>]
    member _.Create([<FromBody>] cmd: CreateFoosballGame) =
        commandQueryDispatcher.SendAsync cmd
        
    [<HttpPost("add-point")>]
    member _.AddPoint([<FromBody>] cmd: AddPointForTeam) =
        commandQueryDispatcher.SendAsync cmd